### DISCUSSION TITLE
How do I redirect www to non-www on an Apache app?

### DISCUSSION THREAD

**Discussion by soycedric** [caprover/caprover#1897] [Q&A]
I want to redirect all traffic that starts with www to my domain without www.  What do I have to do to achieve this?

This is my captain-definition file.

`{
  "schemaVersion": 2,
  "dockerfileLines": [
    "FROM php:8.1-apache",
    "COPY ./public/ /var/www/html/",
    "RUN cp \"$PHP_INI_DIR/php.ini-production\" \"$PHP_INI_DIR/php.ini\""
  ]
}
`

**Comment by githubsaturn** [comment]
There is nothing specific about www  and/or apache.

Just add your non-www domain as a custom domain, and set the redirect to that domain.

![image](https://github.com/caprover/caprover/assets/10582082/5ce50ab4-b09b-42cd-92ce-76b4016f14a8)

**Comment by soycedric** [comment]
This is what worked for me. I put it on top of the nginx configuration.
```

server {
    listen       80;
    server_name  www.<%-s.publicDomain%>;
    return 301 http://<%-s.publicDomain%>$request_uri; }

```
![imagen](https://github.com/caprover/caprover/assets/65123470/a093c014-793a-4249-ba08-b08ade6f6763)
